ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/command/toolbar/selector/PanelSelectorPresenter.java

```diff
@@ -0,0 +1,189 @@
+    public void onSelectorFullEditorClicked() {
+        showPanels(false, false, false);
+    }
+
+    @Override
+    public void onSelectorBottomClicked() {
+        showPanels(false, true, false);
+    }
+
+    @Override
+    public void onSelectorRightClicked() {
+        showPanels(false, false, true);
+    }
+
+    @Override
+    public void onSelectorLeftRightBottomClicked() {
+        showPanels(true, true, true);
+    }
+
+    /**
+     * Sets new visibility for left, bottom and right panels.
+     *
+     * @param left
+     *          left panel
+     * @param bottom
+     *          bottom panel
+     * @param right
+     *          right panel
+     */
+    private void showPanels(boolean left, boolean bottom, boolean right) {
+        Perspective perspective = perspectiveManager.getActivePerspective();
+        if (perspective == null) {
+            return;
+        }
+
+        PartStack editorPartStack = perspective.getPartStack(PartStackType.EDITING);
+        editorPartStack.restore();
+
+        PartStack leftPartStack = perspective.getPartStack(PartStackType.NAVIGATION);
+        PartStack bottomPartStack = perspective.getPartStack(PartStackType.INFORMATION);
+        PartStack rightPartStack = perspective.getPartStack(PartStackType.TOOLING);
+
+        if (left) {
+            if (PartStack.State.MINIMIZED == leftPartStack.getPartStackState()) {
+                leftPartStack.unMinimize();
+            } else {
+                leftPartStack.restore();
+            }
+        } else {
+            leftPartStack.minimize();
+        }
+
+        if (bottom) {
+            if (PartStack.State.MINIMIZED == bottomPartStack.getPartStackState()) {
+                bottomPartStack.unMinimize();
+            } else {
+                bottomPartStack.restore();
+            }
+        } else {
+            bottomPartStack.minimize();
+        }
+
+        if (right) {
+            if (PartStack.State.MINIMIZED == rightPartStack.getPartStackState()) {
+                rightPartStack.unMinimize();
+            } else {
+                rightPartStack.restore();
+            }
+        } else {
+            rightPartStack.minimize();
+        }
+
+        updateButtonState();
+    }
+
+    /**
+     * Updates icon for panel selector button displaying the current state of panels.
+     */
+    private void updateButtonState() {
+        Perspective perspective = perspectiveManager.getActivePerspective();
+        if (perspective == null) {
+            return;
+        }
+
+        PartStack leftPartStack = perspective.getPartStack(PartStackType.NAVIGATION);
+        PartStack bottomPartStack = perspective.getPartStack(PartStackType.INFORMATION);
+        PartStack rightPartStack = perspective.getPartStack(PartStackType.TOOLING);
+
+        if (leftPartStack == null || bottomPartStack == null || rightPartStack == null) {
+            return;
+        }
+
+        if (PartStack.State.MINIMIZED != leftPartStack.getPartStackState() &&
```

Comment:
Could you please refactor this (static import, variables, etc.) to make it possible to read and understand these lines?